app/models/notification.rb

```diff
@@ -4,11 +4,11 @@
 # This class' responsibility is, given a notification blueprint, to determine
 # who are the notification recipients
 class Notification < ApplicationRecord
-  AUDIENCE_USER     = 'user'
-  AUDIENCE_GROUP    = 'usergroup'
-  AUDIENCE_TAXONOMY = 'taxonomy'
-  AUDIENCE_GLOBAL   = 'global'
-  AUDIENCE_ADMIN    = 'admin'
+  AUDIENCE_USER      = 'user'
+  AUDIENCE_USERGROUP = 'usergroup'
+  AUDIENCE_SUBJECT   = 'subject'
```

Comment:
Adding a comment explaining that `AUDIENCE_SUBJECT` works by implementing a method in the subject `notification_recipients_ids` would be nice. 

Long term I would expect to just call a service which figures out what are the 'notification recipients' of an object, to separate concerns better. I don't think it's required just yet since it's just 2 class implementing this, your call. It'd be simpler to understand without the need of adding a comment even. Something like `NotificationRecipientSearcher.new(subject).recipient_ids` would work, and it could even accept parameters.